### DISCUSSION TITLE
Response body is empty

### DISCUSSION THREAD

**Discussion by mattias-wi** [abhinavsingh/proxy.py#1422] [Q&A]
With the plugin below, I'm trying to add a header to the response. But after response.parse returns the buffer seems to be empty. Done some logging in the parse method and just before 'more' becomes False, there is response body data in 'raw'. But not after parse returns.

It works for HTTP GET but not for POST.
Am I missing something or doing something wrong?

```
class AddResponseHeadersPlugin(HttpProxyBasePlugin):

    def handle_upstream_chunk(self, chunk: memoryview) -> None:
        response = HttpParser(httpParserTypes.RESPONSE_PARSER)
        response.parse(chunk)

        if response.is_complete:
            response.add_header(b"Via", b"test")
            self.client.queue(memoryview(response.build_response()))
```

This line seems to makes prevent self.state to get chunkParserStates.COMPLETE and therefore the _process_body method will never assign self.body.

https://github.com/abhinavsingh/proxy.py/blob/a7077cf8db3bb66a6667a9d968a401e8f805e092/proxy/http/parser/chunk.py#L66

**Comment by abhinavsingh** [comment]
@mattias-wi I don't believe `HttpParser` might be able to parse a single standalone chunk if its not a complete chunk.   Chunk parser expects certain end characters to reach COMPLETE state and that might not be happening for your scenarios.   Can you share the chunk that is being passed into the parser?

**Reply by mattias-wi** [reply]
Logged with `raw.tobytes` to see what's passed into  `self.chunk.parse(raw)`:

`b'77\r\n{\n  "apiVersion": "1.30",\n  "method": "getSupportedVersions",\n  "data": {\n    "apiVersions": [\n      "1.30"\n    ]\n  }\n}\r\n'`

When `self.chunk.parse` returns, `self.chunk.body` is:
`b'{\n  "apiVersion": "1.30",\n  "method": "getSupportedVersions",\n  "data": {\n    "apiVersions": [\n      "1.30"\n    ]\n  }\n}'`

and `self.chunk.state` is `WAITING_FOR_SIZE`

**Reply by mattias-wi** [reply]
In my case `self.size` will never be zero before `len(raw) > 0` is False. And therefor `self.state = chunkParserStates.COMPLETE` will never be done.
